### DISCUSSION TITLE
Swapped esc and caps don't work

### DISCUSSION THREAD

**Discussion by franchesoni** [community/community#38944] [Codespaces]
### Select Topic Area

Bug

### Body

i use vim with esc and caps swapped. Even though the swapping is consistent across everything I have tried (web or local), it doesn't work in github codspaces, and pressing Esc would exit Vim edit mode _and_ activate the caps. Pressing caps does nothing.

How can I solve this?

**Comment by erikaellr** [comment]
Hi @franchesoni, we're happy you're using Codespaces! 

I also use vim with esc and caps swapped, and confirmed that this does work for me in the couple of codespaces I just tested it in. But I'm swapping those keys at the OS level on my local machine (with Karabiner for MacOS, if you're on Windows there are similar programs like AutoHotkey), so they get translated before entering the ssh tunnel (that `gh cs ssh` set up for me) I'm using to edit files with vim on the codespaces, and I haven't run into any issues like this with that strategy so far. 

Can you provide more detail about how you're swapping esc and caps?

**Comment by DSDmark** [comment]
Hi @franchesoni,

Try this:-
- go inside your **codespace** and pass ```ctrl+,``` to open the settings.
- after that, click on that icon which is in your right-hand side.

![2022-11-15_07-25](https://user-images.githubusercontent.com/85744816/201811664-e9e9dedd-3d11-4d38-8365-697973ecaf00.png)

- then past that particular code for your **vscode-vim** keybindings.

```
 "vim.easymotion": true,
  "vim.incsearch": true,
  "vim.useSystemClipboard": true,
  "vim.useCtrlKeys": true,
  "vim.hlsearch": true,
  "vim.insertModeKeyBindings": [
    {
      "before": ["j", "j"],
      "after": ["<Esc>"]
    }
  ],
  "vim.normalModeKeyBindingsNonRecursive": [
    {
      "before": ["<leader>", "d"],
      "after": ["d", "d"]
    },
    {
      "before": ["<C-n>"],
      "commands": [":nohl"]
    }
  ],
  "vim.leader": "<space>",
  "vim.handleKeys": {
    "<C-a>": false,
    "<C-f>": false
  },
  "vim.autoindent": false,
```
 
Now You can customize key which you want.
Here for more details :- 
[**Vim-vscode**](https://github.com/VSCodeVim/Vim)

#### or

You could search in settings, there is lots of option to customize vim

![image](https://user-images.githubusercontent.com/85744816/201814656-a58083d7-05ba-4cd3-8452-ef573532c1fe.png)
